scenes/Restaurant/CustomerLoader.gd

```diff
@@ -0,0 +1,323 @@
+	{"line_rgb": "6c4331", "body_rgb": "f57e7d", "eye_rgb": "7ac252 e9f4dc", "horn_rgb": "f1e398"}, # light red
+	{"line_rgb": "6c4331", "body_rgb": "f9bb4a", "eye_rgb": "f9a74c fff6df", "horn_rgb": "b47922"}, # orange
+	{"line_rgb": "6c4331", "body_rgb": "8fea40", "eye_rgb": "f5d561 fcf3cd", "horn_rgb": "b47922"}, # light green
+	{"line_rgb": "6c4331", "body_rgb": "feceef", "eye_rgb": "ffddf4 ffffff", "horn_rgb": "ffffff"}, # pink
+	{"line_rgb": "6c4331", "body_rgb": "b1edee", "eye_rgb": "c1f1f2 ffffff", "horn_rgb": "ffffff"}, # cyan
+	{"line_rgb": "6c4331", "body_rgb": "f9f7d9", "eye_rgb": "91e6ff ffffff", "horn_rgb": "ffffff"}, # white
+	{"line_rgb": "3c3c3d", "body_rgb": "1a1a1e", "eye_rgb": "b8260b f45e40", "horn_rgb": "282828"}, # black
+	{"line_rgb": "6c4331", "body_rgb": "7a8289", "eye_rgb": "f5f0d1 ffffff", "horn_rgb": "282828"}, # grey
+	{"line_rgb": "41281e", "body_rgb": "0b45a6", "eye_rgb": "fad541 ffffff", "horn_rgb": "282828"}  # dark blue
+]
+
+# pool of threads which are currently loading resources
+var _threads := [];
+
+# list of customer definitions which are loaded, but haven't been asked about
+var _ready_customer_definitions := {}
+
+# mutex which ensures we don't load multiple resources concurrently
+var _load_mutex = Mutex.new()
+
+"""
+Loads all the appropriate resources and property definitions for a customer in a background thread. This involves
+loading 20-30 textures and animations, many of which are quite large. The resulting textures are stored back in the
+'customer_def' parameter which is passed in.
+
+Loading these textures can take several seconds. Progress can be monitored by calling 'is_customer_ready'.
+
+Parameter: 'customer_def' describes some high-level information about the customer's appearance, such as 'she has red
+	eyes'. The response includes granular information such as 'her Eye/Sprint/TxMap/RGB value is ff3030'.
+"""
+func summon_customer(customer_def: Dictionary) -> void:
+	_prune_unused_threads()
+	var thread := Thread.new()
+	_threads.append(thread)
+	thread.start(self, "_load_all", customer_def)
+
+"""
+Returns 'true' if the specified customer_def has been fully populated with the necessary textures and metadata to
+draw the customer.
+
+After returning true, this method also purges the customer_def from its cache.
+
+Parameter: 'customer_def' is a customer definition which has been previously passed to the 'summon_customer' method.
+"""
+func is_customer_ready(customer_def: Dictionary) -> bool:
+	return _ready_customer_definitions.erase(customer_def)
+
+"""
+Waits for all threads to finish. Threads must be disposed (or 'joined') for portability.
+"""
+func wait_to_finish():
+	for thread in _threads:
+		thread.wait_to_finish()
+
+func _exit_tree():
+	# wait until all threads are done to dispose of them
+	wait_to_finish()
+
+"""
+Prunes terminated threads from our thread pool.
+"""
+func _prune_unused_threads():
```

Comment:
`unused` and `terminated` are both slightly inaccurate, and it would be better just to say `inactive` since that's what the thread method is called

Reply:
Changed to `prune_inactive_threads`. Also changed comment.